src/Promitor.Core.Scraping/Configuration/Serialization/Deserializer.cs

```diff
@@ -31,11 +68,106 @@ public List<TObject> Deserialize(YamlSequenceNode nodes)
+        }
+        
+        protected void MapOptional<TReturn>(Expression<Func<TObject, TReturn>> accessorExpression, TReturn defaultValue = default, Func<string, IErrorReporter, KeyValuePair<YamlNode, YamlNode>, object> customMapperFunc = null)
+        {
+            var memberExpression = (MemberExpression)accessorExpression.Body;
+            _fields[GetName(memberExpression)] = new FieldContext(memberExpression.Member as PropertyInfo, false, defaultValue, customMapperFunc);
+        }
+
+        private static string GetName(MemberExpression memberExpression)
+        {
+            return Char.ToLowerInvariant(memberExpression.Member.Name[0]) + memberExpression.Member.Name.Substring(1);
+        }
+
+        private static object GetFieldValue(
+            KeyValuePair<YamlNode, YamlNode> fieldNodePair, FieldContext fieldContext, IErrorReporter errorReporter)
+        {
+            if (fieldContext.CustomMapperFunc != null)
+            {
+                return fieldContext.CustomMapperFunc(fieldNodePair.Value.ToString(), errorReporter, fieldNodePair);
+            }
+
+            var propertyType = Nullable.GetUnderlyingType(fieldContext.PropertyInfo.PropertyType) ?? fieldContext.PropertyInfo.PropertyType;
+            if (propertyType.IsEnum)
+            {
+                var parseSucceeded = System.Enum.TryParse(
+                    propertyType, fieldNodePair.Value.ToString(), out var enumValue);
+
+                if (!parseSucceeded)
+                {
+                    errorReporter.ReportError(fieldNodePair.Value, $"'{fieldNodePair.Value}' is not a valid value for '{fieldNodePair.Key}'.");
+                }
+
+                return enumValue;
+            }
+            
+            if (typeof(IDictionary<string, string>).IsAssignableFrom(propertyType))
+            {
+                return ((YamlMappingNode)fieldNodePair.Value).GetDictionary();
+            }
+
+            if (propertyType == typeof(TimeSpan))
+            {
+                var parseSucceeded = TimeSpan.TryParse(fieldNodePair.Value.ToString(), out var timeSpanValue);
+                if (!parseSucceeded)
+                {
+                    errorReporter.ReportError(fieldNodePair.Value, $"'{fieldNodePair.Value}' is not a valid value for '{fieldNodePair.Key}'. The value must be in the format 'hh:mm:ss'.");
+                }
+
+                return timeSpanValue;
+            }
+
+            try
+            {
+                return Convert.ChangeType(fieldNodePair.Value.ToString(), fieldContext.PropertyInfo.PropertyType);
+            }
+            catch (FormatException)
+            {
+                errorReporter.ReportError(fieldNodePair.Value, $"'{fieldNodePair.Value}' is not a valid value for '{fieldNodePair.Key}'. The value must be of type {fieldContext.PropertyInfo.PropertyType}.");
+            }
+
+            return null;
+        }
+
+        private class FieldContext
```

Comment:
For final PR, please move to dedicated file

Reply:
Will do.